src/libs/ajax/leonardo/Disks.test.ts

```diff
@@ -36,4 +58,77 @@ describe('Disks ajax', () => {
     expect(mockFetchLeo).toHaveBeenCalledWith('api/google/v1/disks', expect.anything());
     expect(disks).toStrictEqual(rawJson);
   });
+
+  it('should call disk delete v1 endpoint with proper auth functions', async () => {
+    // Arrange
+    const disk = galaxyDisk;
+    const mockFetchLeo = jest.fn();
+    asMockedFn(fetchLeo).mockImplementation(mockFetchLeo);
+
+    // Act
+    await Disks(signal).disksV1().disk(disk.cloudContext.cloudResource, disk.name).delete();
+
+    // Assert
+    expect(mockWatchWithAuth).toBeCalledTimes(1);
+    expect(mockWatchWithAppId).toBeCalledTimes(1);
+    expect(mockFetchLeo).toHaveBeenCalledWith(`api/google/v1/disks/${disk.cloudContext.cloudResource}/${disk.name}`, {
+      signal,
+      method: 'DELETE',
+    });
+  });
+
+  it('should call disk update v1 endpoint with proper auth functions', async () => {
+    // Arrange
+    const disk = galaxyDisk;
+    const size = 100;
+    const mockFetchLeo = jest.fn();
+    asMockedFn(fetchLeo).mockImplementation(mockFetchLeo);
+
+    // Act
+    await Disks(signal).disksV1().disk(disk.cloudContext.cloudResource, disk.name).update(size);
+
+    // Assert
+    expect(mockWatchWithAuth).toBeCalledTimes(1);
+    expect(mockWatchWithAppId).toBeCalledTimes(1);
+    expect(mockFetchLeo).toHaveBeenCalledWith(`api/google/v1/disks/${disk.cloudContext.cloudResource}/${disk.name}`, {
+      signal,
+      method: 'PATCH',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ size }),
+    });
+  });
+
+  it('should call the details disk v1 endpoint', async () => {
+    // Arrange
+    const disk = galaxyDisk;
+    const mockResFn = jest.fn().mockReturnValue(Promise.resolve(undecoratePd(galaxyDisk)));
+    const mockRes = { json: mockResFn };
+    const mockFetchLeo = jest.fn().mockReturnValue(Promise.resolve(mockRes));
+    asMockedFn(fetchLeo).mockImplementation(mockFetchLeo);
+
+    // Act
+    const details = await Disks(signal).disksV1().disk(disk.cloudContext.cloudResource, disk.name).details();
+
+    // Assert
+    expect(mockFetchLeo).toHaveBeenCalledWith(`api/google/v1/disks/${disk.cloudContext.cloudResource}/${disk.name}`, {
+      signal,
+      method: 'GET',
+    });
+    expect(details).toStrictEqual(undecoratePd(galaxyDisk));
+  });
+
+  it('should call the delete disk v2 endpoint', async () => {
```

Comment:
Quick question, why is there no need to test the V2 endpoint only for delete?

Reply:
thats the only method that we call on the v2 disk endpoint!

Reply:
Oh weird, thanks for clarifying 👍